src/main/kotlin/com/github/zly2006/enclosure/EnclosureArea.kt

```diff
@@ -153,8 +153,9 @@ open class EnclosureArea : PersistentState, EnclosureView {
         val y = world.getTopY(Heightmap.Type.MOTION_BLOCKING_NO_LEAVES, x, z)
         if (y == world.bottomY) {
             // this player is alive, but in a void
-            minecraftServer.playerManager.respawnPlayer(player, true)
-            minecraftServer.overworld.chunkManager.updatePosition(player)
+            val overworld = player.server.overworld
+            val spawnPos = overworld.spawnPos
+            player.teleport(overworld, spawnPos.x.toDouble(), spawnPos.y.toDouble(), spawnPos.z.toDouble(), 0f, 0f)
```

Comment:
you should +0.5 for block pos on x/z so that the player will tp to the center of the block